doc/HOWTO/howto_readState.md

```diff
@@ -0,0 +1,165 @@
+      contract: new Address(contractId).toScAddress(),
+      key: xdr.ScVal.scvSymbol(symbolText),
+      durability: xdr.ContractDataDurability.persistent(),
+    }),
+  );
+  return ledgerKey.toXDR("base64");
+};
+```
+
+We need to call `getLedgerKeySymbol(contractID, "COUNTER")`. If we had a contract with multiple fields, we would call this method for each of them:
+```
+k = getLedgerKeySymbol(contractID, "COUNTER")
+k_foo = getLedgerKeySymbol(contractID, "FOO")
+k_bar = getLedgerKeySymbol(contractID, "BAR")
+```
+
+With the key, or set of keys, in hand, we now invoke the RPC, to querry the current values of the fields these keys represent.
+
+In Javascript, querying the RPC is done as follows:
+```js
+let res = await fetch(NET, {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(requestBody),
+})
+let json = await res.json()
+```
+where `NET` is the Soroban network being queried, for example 
+`'https://soroban-testnet.stellar.org'`, and `requestBody` is the specific request we're making of the network.
+
+Since we aim to read a ledger entry, our request will use the `getLedgerEntries` method, and `requestBody` will look like this:
+```js
+let requestBody = {
+  "jsonrpc": "2.0",
+  "id": ...,
+  "method": "getLedgerEntries",
+  "params": {
+    "keys": [
+      k
+    ]
+  }
+}
+```
+
+Here, if we had multiple fields to read, we would simply use
+```js
+...
+"keys": [
+      k, k_foo, k_bar
+    ]
+```
+
+Our result would look like this:
+```js
+{
+  "jsonrpc": "2.0",
+  "id": ...,
+  "result": {
+    "entries": [
+      {
+        "key": ...,
+        "xdr": XDR,
+        "lastModifiedLedgerSeq": ...
+      }
+    ],
+    "latestLedger": ...
+  }
+}
+```
+
+where the value of the field that we seek is contained within `XDR`. With multiple keys, the difference would simply be
+```js
+...
+"entries": [
+      {
+        "key": ...,
+        "xdr": XDR,
+        "lastModifiedLedgerSeq": ...
+      },
+      {
+        "key": ...,
+        "xdr": XDR_FOO,
+        "lastModifiedLedgerSeq": ...
+      }
+      {
+        "key": ...,
+        "xdr": XDR_BAR,
+        "lastModifiedLedgerSeq": ...
+      }
+    ],
+```
+
+The final step is unpacking the obtained XDR, from which we  read the value associated with our key. This [functionality][sdk] is available from the `stellar-sdk` (`LedgerEntryData.from_xdr`), as well as [online][lab] for quick experimentation.
+
+Reading the `XDR` value, we should end up with the following shape:
+```
+LedgerEntryData: [contractData]
+    contractData
+        ext: [undefined]
+        contract: [scAddressTypeContract]
+            contractId: ...
+        key: [scvVec]
+            vec: Array[2]
+                [0]: [scvSymbol]
+                    sym: Counter
+                [1]: [scvAddress]
+                    address: ...
+        durability: {"name":"persistent","value":1}
+        val: [scvU32]
+            u32: VAL
```

Comment:
It looks here like `VAL` is keyed by it's type?

Can you include how we would read `VAL` from this, given either the XDR or whatever the stellar SDK translates it to (I assume JSON)?